internal/clients/compute/group/group.go

```diff
@@ -0,0 +1,410 @@
+		return false
+	case observed.Properties.AccessAndManageMonitoring != nil && *observed.Properties.AccessAndManageMonitoring != cr.Spec.ForProvider.AccessAndManageMonitoring:
+		return false
+	case observed.Properties.CreateBackupUnit != nil && *observed.Properties.CreateBackupUnit != cr.Spec.ForProvider.CreateBackupUnit:
+		return false
+	case observed.Properties.CreateDataCenter != nil && *observed.Properties.CreateDataCenter != cr.Spec.ForProvider.CreateDataCenter:
+		return false
+	case observed.Properties.CreateFlowLog != nil && *observed.Properties.CreateFlowLog != cr.Spec.ForProvider.CreateFlowLog:
+		return false
+	case observed.Properties.CreateInternetAccess != nil && *observed.Properties.CreateInternetAccess != cr.Spec.ForProvider.CreateInternetAccess:
+		return false
+	case observed.Properties.CreateK8sCluster != nil && *observed.Properties.CreateK8sCluster != cr.Spec.ForProvider.CreateK8sCluster:
+		return false
+	case observed.Properties.CreatePcc != nil && *observed.Properties.CreatePcc != cr.Spec.ForProvider.CreatePcc:
+		return false
+	case observed.Properties.CreateSnapshot != nil && *observed.Properties.CreateSnapshot != cr.Spec.ForProvider.CreateSnapshot:
+		return false
+	case observed.Properties.ManageDBaaS != nil && *observed.Properties.ManageDBaaS != cr.Spec.ForProvider.ManageDBaaS:
+		return false
+	case observed.Properties.ManageDataplatform != nil && *observed.Properties.ManageDataplatform != cr.Spec.ForProvider.ManageDataPlatform:
+		return false
+	case observed.Properties.ManageRegistry != nil && *observed.Properties.ManageRegistry != cr.Spec.ForProvider.ManageRegistry:
+		return false
+	case observed.Properties.ReserveIp != nil && *observed.Properties.ReserveIp != cr.Spec.ForProvider.ReserveIP:
+		return false
+	case observed.Properties.S3Privilege != nil && *observed.Properties.S3Privilege != cr.Spec.ForProvider.S3Privilege:
+		return false
+	}
+	configuredMemberIDs := memberIDsSet(cr)
+	observedMemberIDs := sets.New[string](cr.Status.AtProvider.UserIDs...)
+	if !observedMemberIDs.Equal(configuredMemberIDs) {
+		return false
+	}
+
+	configuredResourceShares := resourceSharesSet(cr)
+	observedResourceShares := sets.New[v1alpha1.ResourceShare](cr.Status.AtProvider.ResourceShares...)
+	return observedResourceShares.Equal(configuredResourceShares)
+}
+
+func memberIDsSet(cr *v1alpha1.Group) sets.Set[string] {
+	mCount := len(cr.Spec.ForProvider.UserCfg)
+	memberIDs := sets.Set[string]{}
+	for i := 0; i < mCount; i++ {
+		memberIDs.Insert(cr.Spec.ForProvider.UserCfg[i].UserID)
+	}
+	return memberIDs
+}
+
+func resourceSharesSet(cr *v1alpha1.Group) sets.Set[v1alpha1.ResourceShare] {
+	rsCount := len(cr.Spec.ForProvider.ResourceShareCfg)
+	resourceShares := sets.Set[v1alpha1.ResourceShare]{}
+	ids := sets.Set[string]{}
+	for i := 0; i < rsCount; i++ {
+		resourceShareID := cr.Spec.ForProvider.ResourceShareCfg[i].ResourceID
+		if resourceShareID != "" && !ids.Has(resourceShareID) {
```

Comment:
If duplicates is a possibility, may be we need to return an error.  most likely the duplication will be because of user mistake, but it may save the user some debugging time if we returned a duplication error.